content/tutorials/permissionless-paymaster/10.index.md

```diff
@@ -0,0 +1,306 @@
+
+```solidity
+        bytes32 messageHash = keccak256(
+            abi.encode(
+                SIGNATURE_TYPEHASH,
+                _from,
+                _to,
+                _expirationTime,
+                _maxNonce,
+                _maxFeePerGas,
+                _gasLimit
+            )
+        );
+```
+
+- Create a `getSignature()` function in your backend(recommended)
+
+```javascript
+// Example code
+import {BigNumber, Contract, Wallet} from "zksync-ethers";
+export async function getSignature(
+  from: string, to: string, expirationTime: BigNumber, maxNonce: BigNumber, maxFeePerGas: BigNumber, gasLimit: BigNumber, paymaster: Contract
+){
+  const signer = new Wallet(process.env.SIGNER_PRIVATE_KEY, provider);
+// EIP-712 domain from the paymaster
+  const eip712Domain = await paymaster.eip712Domain();
+  const domain = {
+    name: eip712Domain[1],
+    version: eip712Domain[2],
+    chainId: eip712Domain[3],
+    verifyingContract: eip712Domain[4],
+  }
+  const types = {
+    PermissionLessPaymaster: [
+      { name: "from", type: "address"},
+      { name: "to", type: "address"},
+      { name: "expirationTime", type: "uint256"},
+      { name: "maxNonce", type: "uint256"},
+      { name: "maxFeePerGas", type: "uint256"},
+      { name: "gasLimit", type: "uint256"}
+    ]
+  };
+// -------------------- IMPORTANT --------------------
+  const values = {
+    from,  // User address
+    to, // Your dapp contract address which the user will interact
+    expirationTime, // Expiration time post which the signature expires
+    maxNonce, // Max nonce of user after which signature becomes invalid
+    maxFeePerGas, // Current max gas price
+    gasLimit // Max gas limit you want to allow to your user. Ensure to add 60K gas for paymaster overhead.
+  }
+// Note: MaxNonce allows the signature to be replayed.
+// For eg: If currentNonce of user is 5, maxNonce is set to 10. Signature will allowed to replayed for nonce 6,7,8,9,10 on the same `to` address by the same user.
+// This is to provide flexibility to Dapps to ensure signature works if users have multiple transactions running. 
+// Important: Signers are recommended to set maxNonce as current nonce of the user or as close as possible to ensure safety of gas funds.
+// Important : Signers should set expirationTime is close enough to ensure safety of funds.
+
+// Signer wallet will already defined in the code
+  return [(await signer._signTypedData(domain, types, values)), signer.address];
+}
+```
+
+## 4. Add paymaster data and signature into transaction blob
+
+- Before sending user transaction, you will need to add customData related to paymaster.
+- Funds are deducted as per : `gasPrice`*`gasLimit`. Dapp should ensure enough funds are deposited to avoid failures.
+- Note: For simplicity purposes, we use "DappContract" variable which is expected to be defined as per Dapp's need.
+
+```javascript
+// This is example code. Direct copy/paste won't work
```

Comment:
Can you explain more why this code won't work if you copy/paste it and what exactly needs to be changed to make it work?